pkg/analysis/completion.go

```diff
@@ -142,6 +126,43 @@ func (a *Analyzer) completeExpression(doc document.Document, nodes []*sitter.Nod
 	return symbols
 }
 
+// Returns a list of available symbols for completion as follows:
+// - If in a function argument list, include keyword args for that function
+// - Add symbols in scope for the node at point
+// - Add document symbols, excluding symbols after the node if the node is in the module scope
```

Comment:
Does this 9173887bd5add7c1b886b6e917b33d8b5aefc65f help?

Reply:
Ah, no. Added more explanation in 03e1e133d5302ffb505245a6f267164fe7e8d212. Not sure how much explanation is necessary in comments, but what I mean is this. Consider:
```python
a = 1
# <- point 1
def b():
  # <- point 2
  return 2
b()
# <- point 3
c = 3
```
1. At point 1, the visible symbols should only be `a`.
2. At point 2, the visible symbols should be `a`, `b`, and `c` since we're inside a function.
3. At point 3, the visible symbols should be `a` and `b`, but not `c`.

Not sure what the Starlark spec says about scoping but this is the behavior I have noticed. Let me know if you think that's not quite right.

Reply:
yeah, if you enter this at point 3, it prints 3:
```
f = lambda: print(c)
f()
```

Maybe a lambda is actually the only exception to that and we could special case it? I'd think I'd rather simplify and cut the special case entirely rather than add a special case within a special case.

fwiw, the python extension just completes it:
<img width="392" alt="image" src="https://user-images.githubusercontent.com/7453991/162455331-87943edc-995e-4eb5-b092-a490fa64b169.png">

The starlark spec says it's a static error to refer to a variable name that's not bound at all, and a dynamic error to refer to a variable name before it's bound